front/pages/api/w/[wId]/data_sources/[name]/databases/csv.ts

```diff
@@ -0,0 +1,324 @@
+import { parse } from "csv-parse";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSource } from "@app/lib/api/data_sources";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { CoreAPI, CoreAPIDatabaseRow } from "@app/lib/core_api";
+import { isDevelopmentOrDustWorkspace } from "@app/lib/development";
+import { APIError } from "@app/lib/error";
+import { Err, Ok, Result } from "@app/lib/result";
+import { generateModelSId } from "@app/lib/utils";
+import logger from "@app/logger/logger";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+const CreateDatabaseFromCsvSchema = t.type({
+  name: t.string,
+  description: t.string,
+  csv: t.string,
+});
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  if (!owner || !plan || !auth.isBuilder()) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  if (!isDevelopmentOrDustWorkspace(owner)) {
+    res.status(404).end();
+    return;
+  }
+
+  const dataSource = await getDataSource(auth, req.query.name as string);
+  if (!dataSource) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST":
+      const bodyValidation = CreateDatabaseFromCsvSchema.decode(req.body);
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+        return apiError(req, res, {
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body: ${pathError}`,
+          },
+          status_code: 400,
+        });
+      }
```

Comment:
Can we put a limit on size of data (say 1mb) :pray:

Reply:
There's already such a limit on the frontend side (1 million characters). Would you prefer that I add another one here ?